src/contracts/Model/Suggestion/SuggestionCollection.php

```diff
@@ -47,4 +56,19 @@ public function truncate(int $count): void
     {
         $this->items = array_slice($this->items, 0, $count);
     }
+
+    public function increaseTotalCount(int $totalCount): void
```

Comment:
Wouldn't increasing `totalCount' when using `append' be less error prone?

Reply:
This method is needed because of multiple data sources. When the `limit` value (for example, 5) is passed, only 5 records will be inserted using the `append` method. However, it is necessary to include the total number of records that the query finds.

Reply:
yeah, I realised that this is probably the case later in the day, wonder how it will work with pagination tho (offset), but I guess this is not something we need (or at all if this works properly), now.